src/back/hlsl/writer.rs

```diff
@@ -1635,76 +1635,105 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
             } => {
                 use crate::MathFunction as Mf;
 
-                let fun_name = match fun {
-                    // comparison
-                    Mf::Abs => "abs",
-                    Mf::Min => "min",
-                    Mf::Max => "max",
-                    Mf::Clamp => "clamp",
-                    // trigonometry
-                    Mf::Cos => "cos",
-                    Mf::Cosh => "cosh",
-                    Mf::Sin => "sin",
-                    Mf::Sinh => "sinh",
-                    Mf::Tan => "tan",
-                    Mf::Tanh => "tanh",
-                    Mf::Acos => "acos",
-                    Mf::Asin => "asin",
-                    Mf::Atan => "atan",
-                    Mf::Atan2 => "atan2",
-                    // decomposition
-                    Mf::Ceil => "ceil",
-                    Mf::Floor => "floor",
-                    Mf::Round => "round",
-                    Mf::Fract => "frac",
-                    Mf::Trunc => "trunc",
-                    Mf::Modf => "modf",
-                    Mf::Frexp => "frexp",
-                    Mf::Ldexp => "ldexp",
-                    // exponent
-                    Mf::Exp => "exp",
-                    Mf::Exp2 => "exp2",
-                    Mf::Log => "log",
-                    Mf::Log2 => "log2",
-                    Mf::Pow => "pow",
-                    // geometry
-                    Mf::Dot => "dot",
-                    //Mf::Outer => ,
-                    Mf::Cross => "cross",
-                    Mf::Distance => "distance",
-                    Mf::Length => "length",
-                    Mf::Normalize => "normalize",
-                    Mf::FaceForward => "faceforward",
-                    Mf::Reflect => "reflect",
-                    Mf::Refract => "refract",
-                    // computational
-                    Mf::Sign => "sign",
-                    Mf::Fma => "fma",
-                    Mf::Mix => "lerp",
-                    Mf::Step => "step",
-                    Mf::SmoothStep => "smoothstep",
-                    Mf::Sqrt => "sqrt",
-                    Mf::InverseSqrt => "rsqrt",
-                    //Mf::Inverse =>,
-                    Mf::Transpose => "transpose",
-                    Mf::Determinant => "determinant",
-                    // bits
-                    Mf::CountOneBits => "countbits",
-                    Mf::ReverseBits => "reversebits",
-                    _ => return Err(Error::Unimplemented(format!("write_expr_math {:?}", fun))),
-                };
+                match fun {
+                    Mf::Asinh | Mf::Acosh => {
```

Comment:
Generally we'd want to avoid double-matching on `fun`, because it's repetitive and error-prone.
Instead, we should do something like this:
```
enum Function {
  Asincosh { is_sin: bool },
  Atanh,
  Regular(&'static str),
}
```

This way, we'd construct it in the first match, and then do a match on the result specifically.